This patch is part of a series of patches for speech-tools for
Debian GNU/Linux, changes the default orfer of sound output methods

Index: b/audio/gen_audio.cc
===================================================================
--- a/audio/gen_audio.cc	2011-10-27 09:46:01.000000000 +1100
+++ b/audio/gen_audio.cc	2011-11-11 08:46:56.000000000 +1100
@@ -95,6 +95,10 @@
       protocol = "macosxaudio";
 	else if (win32audio_supported)
 	    protocol = "win32audio";
+	else if (esd_supported)
+	    protocol = "esdaudio";
+	else if (nas_supported)
+	    protocol = "netaudio";
 	else if (mplayer_supported)
 	    protocol = "mplayeraudio";
 	else
@@ -243,10 +247,6 @@
     audios += "sunaudio";  // we always support this in spite of the hardware
 
     audios += " audio_command";
-    if (nas_supported)
-	audios += " netaudio";
-    else if (esd_supported)
-	audios += " esdaudio";
     if (sun16_supported)
 	audios += " sun16audio";
     if (freebsd16_supported)
@@ -285,11 +285,7 @@
 	protocol = sr;
     else if (protocol == "")
     {
-	if (nas_supported)
-	    protocol = "netaudio";  // the default protocol
-	else if (esd_supported)
-	    protocol = "esdaudio";  // the default protocol
-	else if (sun16_supported)
+	if (sun16_supported)
 	    protocol = "sun16audio";
 	else if (freebsd16_supported)
 	    protocol = "freebsd16audio";
@@ -299,6 +295,10 @@
 	    protocol = "irixaudio";
 	else if (win32audio_supported)
 	    protocol = "win32audio";
+	else if (esd_supported)
+	    protocol = "esdaudio";
+	else if (nas_supported)
+	    protocol = "netaudio";
 	else if (mplayer_supported)
 	    protocol = "mplayeraudio";
 	else
